it/src/main/java/com/google/cloud/teleport/it/pubsub/PubsubUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.pubsub;
+
+import com.google.pubsub.v1.Subscription;
+import com.google.pubsub.v1.SubscriptionName;
+import com.google.pubsub.v1.Topic;
+import com.google.pubsub.v1.TopicName;
+
+/** Utilities to make working with Pubsub easier. */
+public final class PubsubUtils {
+  private PubsubUtils() {}
+
+  /**
+   * Convert the instance of {@link Topic} to a {@link TopicName}.
+   *
+   * <p>The idea of this project is to abstract the conversion/parse of the format
+   * <strong>projects/{project_id}/topics/{topic_name}</strong>
+   *
+   * @param topic Base topic.
+   * @return The reference to the topic.
+   */
+  public static TopicName toTopicName(Topic topic) {
```

Comment:
This seems like an error-prone approach, or can we guarantee that `Topic` always returns the correct format? This should probably use a regex and capture group. (Note: We use RE2J rather than Java's built-in regex library.)

Reply:
Makes sense, thanks!